src/View/Helper/FormHelper.php

```diff
@@ -276,4 +302,122 @@ protected function _parseOptions($fieldName, $options)
         }
         return $options;
     }
+
+    /**
+     * Form alignment detector/switcher.
+     *
+     * @param array $options Options.
+     * @return array Modified options.
+     */
+    protected function _formAlignment($options)
+    {
+        if (!$options['align']) {
+            $options['align'] = $this->_detectFormAlignment($options);
+        }
+
+        if (is_array($options['align'])) {
+            $this->_grid = $options['align'];
+            $options['align'] = 'horizontal';
+        } elseif ($options['align'] === 'horizontal') {
+            $this->_grid = $this->config('grid');
+        }
+
+        if (!in_array($options['align'], ['default', 'horizontal', 'inline'])) {
+            throw new InvalidArgumentException('Invalid `align` option value.');
+        }
+
+        $this->_align = $options['align'];
+
+        unset($options['align']);
+
+        if ($this->_align === 'default') {
+            $options['templates'] += [
+               'checkboxContainer' => '<div class="checkbox">{{content}}{{help}}</div>',
+               'checkboxContainerError' => '<div class="checkbox has-error">{{content}}{{error}}{{help}}</div>',
+            ];
+            return $options;
+        }
+
+        $options = $this->injectClasses('form-' . $this->_align, $options);
+
+        if ($this->_align === 'inline') {
+            $options['templates'] += [
+               'label' => '<label class="sr-only"{{attrs}}>{{text}}</label>',
+               'inputContainer' => '{{content}}'
+            ];
+            return $options;
+        }
+
+        $offsetedGridClass = implode(' ', [$this->_gridClass('left', true), $this->_gridClass('middle')]);
+        $options['templates'] += [
```

Comment:
I am still not happy about this. While this allows you to customize the templates through create() or input() calls, one can't do so using config when loading helper. If i do `$this->loadHelper('Form', ['className' => 'BootstrapUI.Form', 'templates' => [...]]` my templates will get overriden by the ones set by this code.

Solution to this would be what i had proposed earlier, having a set of templates per alignment type which can be modified through config and then swapped here based on form alignment type.